src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -48,110 +52,67 @@ void GinIndexPostingsBuilder::add(UInt32 row_id)
 UInt64 GinIndexPostingsBuilder::serialize(WriteBuffer & buffer)
 {
     rowids.runOptimize();
-
     const UInt64 cardinality = rowids.cardinality();
 
-    if (cardinality < MIN_SIZE_FOR_ROARING_ENCODING)
+    PaddedPODArray<UInt32> deltas(cardinality);
```

Comment:
Are we still getting memory corruptions with FastPFOR?

Reply:
no, the case `compressed size > uncompressed size` happened due to a few elements in array, after I added a threshold it didn't occur then